x-pack/plugins/maps/public/connected_components/timeslider/timeslider.tsx

```diff
@@ -0,0 +1,179 @@
+}
+
+// Why Timeslider and KeyedTimeslider?
+// Using react 'key' property to ensure new KeyedTimeslider instance whenever props.timeRange changes
+export function Timeslider(props: Props) {
+  return props.isTimesliderOpen ? (
+    <KeyedTimeslider key={`${props.timeRange.from}-${props.timeRange.to}`} {...props} />
+  ) : null;
+}
+
+class KeyedTimeslider extends Component<Props, State> {
+  private _isMounted: boolean = false;
+
+  constructor(props: Props) {
+    super(props);
+    const timeRangeBounds = getTimeFilter().calculateBounds(props.timeRange);
+    if (timeRangeBounds.min === undefined || timeRangeBounds.max === undefined) {
+      throw new Error(
+        'Unable to create Timeslider component, timeRangeBounds min or max are undefined'
+      );
+    }
+    const min = timeRangeBounds.min.valueOf();
+    const max = timeRangeBounds.max.valueOf();
+    const interval = getInterval(min, max);
+    const timeslice: [number, number] = [min, max];
+
+    this.state = {
+      max,
+      min,
+      range: interval,
+      ticks: getTicks(min, max, interval),
+      timeslice,
+      timesliceText: prettyPrintTimeslice(timeslice),
+    };
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  componentDidMount() {
+    this._isMounted = true;
+    // auto-select range between first tick and second tick
+    this._onChange([this.state.ticks[0].value, this.state.ticks[1].value]);
+  }
+
+  _doesTimesliceCoverTimerange() {
+    return this.state.timeslice[0] === this.state.min && this.state.timeslice[1] === this.state.max;
+  }
+
+  _onDualControlChange = (value: [number | string, number | string]) => {
+    this.setState({ range: (value[1] as number) - (value[0] as number) }, () => {
+      this._onChange(value as [number, number]);
+    });
+  };
+
+  _onChange = (value: [number, number]) => {
+    this.setState({
+      timeslice: value,
+      timesliceText: prettyPrintTimeslice(value),
+    });
+    this._propagateChange(value);
+  };
+
+  _onNext = () => {
+    const from =
+      this._doesTimesliceCoverTimerange() || this.state.timeslice[1] === this.state.max
+        ? this.state.ticks[0].value
+        : this.state.timeslice[1];
+    const to = from + this.state.range;
+    this._onChange([from, to <= this.state.max ? to : this.state.max]);
+  };
+
+  _onPrevious = () => {
+    const to =
+      this._doesTimesliceCoverTimerange() || this.state.timeslice[0] === this.state.min
+        ? this.state.ticks[this.state.ticks.length - 1].value
+        : this.state.timeslice[0];
+    const from = to - this.state.range;
+    this._onChange([from < this.state.min ? this.state.min : from, to]);
+  };
+
+  _propagateChange = _.debounce((value: [number, number]) => {
+    if (this._isMounted) {
+      this.props.setTimeslice({ from: value[0], to: value[1] });
+    }
+  }, 300);
+
+  render() {
+    return (
+      <div className="mapTimeslider">
+        <div className="mapTimeslider__row">
+          <EuiButtonIcon
```

Comment:
Do we need this close button? There's already an on/off button on the left with active/inactive state.

Reply:
It was in the design provided by @miukimiu